### PR TITLE
Trigger next rollout group - backend and management API implementatio…

### PR DESCRIPTION
**Pull request by dimitar-shterev**
Trigger next rollout group - Back-end and management API implementation. Back-end and management API tests.

Signed-off-by: Dimitar Shterev <dimitar.shterev@bosch.io>

### CODE REVIEW

**Comment by StefanKlt** [comment]
Please make sure to test also edge and error cases on management level (not existing rollout, different combinations of rollouts states and group states, ...)

File: hawkbit-rest/hawkbit-mgmt-resource/src/test/java/org/eclipse/hawkbit/mgmt/rest/resource/MgmtRolloutResourceTest.java [hunk 1]
```diff
@@ -1043,4 +1045,57 @@ amountGroups, new RolloutGroupConditionBuilder().withDefaults()
...
+                .andExpect(jsonPath("id", equalTo(rollout.getId().intValue())))
+                .andExpect(jsonPath("status", equalTo("running")));
+
+        final RolloutGroup firstGroup = rolloutGroupManagement
+                .findByRollout(PageRequest.of(0, 1, Direction.ASC, "id"), rollout.getId()).getContent().get(0);
+        final RolloutGroup secondGroup = rolloutGroupManagement
+                .findByRollout(PageRequest.of(1, 1, Direction.ASC, "id"), rollout.getId()).getContent().get(0);
+
+
+        retrieveAndVerifyRolloutGroupInRunning(rollout, firstGroup);
```
Comment (StefanKlt) [inline hawkbit-rest/hawkbit-mgmt-resource/src/test/java/org/eclipse/hawkbit/mgmt/rest/resource/MgmtRolloutResourceTest.java:1083]: There is no need to make rest calls for that. Just check if firstGroup.getStatus() and secondGroup .getStatus() has the expected value

File: hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/RolloutManagement.java [hunk 2]
```diff
@@ -463,4 +463,19 @@ Slice<Rollout> findByFiltersWithDetailedStatus(@NotNull Pageable pageable, @NotE
...
+     * Triggers next group of a rollout for processing even success threshold isn't met yet.
+     * Current running groups will not change their status.
+     *
+     * @param rolloutId
+     *            the rollout to be paused.
+     *
+     * @throws EntityNotFoundException
+     *             if rollout or group with given ID does not exist
+     * @throws RolloutIllegalStateException
+     *             if given rollout is not in {@link RolloutStatus#RUNNING}.
```
[the remark this answers is not in the document]
Comment (StefanKlt) [reply hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/RolloutManagement.java:476]: OK, I guess then we still throw RolloutIllegalStateException

[the remark this answers is not in the document]
Comment (dimitar-shterev) [reply hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/RolloutManagement.java:476]: Added a condition to check if there is any group in scheduling state left.

File: hawkbit-repository/hawkbit-repository-jpa/src/test/java/org/eclipse/hawkbit/repository/jpa/RolloutManagementTest.java [hunk 3]
```diff
@@ -2074,4 +2075,48 @@ private void awaitRunningState(final Long myRolloutId) {
...
+        final String errorCondition = "80";
+
+        final Rollout createdRollout = createAndStartRollout(amountTargetsForRollout, amountOtherTargets, amountGroups,
+                successCondition, errorCondition);
+
+        // triggers next group
+        rolloutManagement.triggerNextGroup(createdRollout.getId());
+
+        //second group should in running state
+        List<RolloutGroup> runningRolloutGroups = rolloutGroupManagement
```
[the remark this answers is not in the document]
Comment (dimitar-shterev) [reply hawkbit-repository/hawkbit-repository-jpa/src/test/java/org/eclipse/hawkbit/repository/jpa/RolloutManagementTest.java:2096]: done

File: hawkbit-rest/hawkbit-mgmt-resource/src/test/java/org/eclipse/hawkbit/mgmt/rest/resource/MgmtRolloutResourceTest.java [hunk 4]
```diff
@@ -1033,14 +1040,95 @@ private void postRollout(final String name, final int groupSize, final Long dist
 
     private Rollout createRollout(final String name, final int amountGroups, final long distributionSetId,
             final String targetFilterQuery) {
-        final Rollout rollout = rolloutManagement.create(
+        final Rollout rollout = createRolloutInCreatingSatate(name, amountGroups, distributionSetId, targetFilterQuery);
```
Comment (herdt-michael) [inline hawkbit-rest/hawkbit-mgmt-resource/src/test/java/org/eclipse/hawkbit/mgmt/rest/resource/MgmtRolloutResourceTest.java:1043]:
```suggestion
        final Rollout rollout = createRolloutInCreatingState(name, amountGroups, distributionSetId, targetFilterQuery);
```

**Review by StefanKlt — APPROVED** [review]
Looks good, thanks @dimitar-shterev

**Pull request merged**